### PR TITLE
move proxy env vars to each provisioner

### PR DESCRIPTION
**Pull request by dan-hill2802**
change conditionals around commas to ensure that any combination of `provision_scripts` and `provision_script_keys` can be set

### CODE REVIEW

**Comment by dan-hill2802** [comment]
Template validates locally

**Comment by dan-hill2802** [comment]
Considered using [use_env_var_file](https://www.packer.io/docs/provisioners/shell.html#use_env_var_file) but it would pass on unwanted env vars from [Lambda](https://docs.aws.amazon.com/lambda/latest/dg/lambda-environment-variables.html)

File: dw-general-ami/generic_packer_template.json.j2 [hunk 1]
```diff
@@ -33,68 +34,49 @@
...
+        {
           "type": "shell",
           "inline": "{{ provisioner_script }}"
+          {% if 'set_proxy' in event and event['set_proxy'] == true %}
+          ,
+          "environment_vars": [
+            "HTTP_PROXY={% raw %}{{ user `http_proxy` }}{% endraw %}",
+            "HTTPS_PROXY={% raw %}{{ user `https_proxy` }}{% endraw %}",
+            "http_proxy={% raw %}{{ user `http_proxy` }}{% endraw %}",
+            "https_proxy={% raw %}{{ user `http_proxy` }}{% endraw %}",
```
Comment (mattburgess) [inline dw-general-ami/generic_packer_template.json.j2]: should be `user https_proxy`

File: dw-general-ami/generic_packer_template.json.j2 [hunk 2]
```diff
@@ -33,68 +34,49 @@
...
           "type": "shell",
           "script": "{{download_dir}}/{{ provisioner }}",
           "execute_command": "sudo -E sh -eux -c '{% raw %}{{ .Path }}{% endraw %}'"
+          {% if 'set_proxy' in event and event['set_proxy'] == true %}
+          ,
+          "environment_vars": [
+            "HTTP_PROXY={% raw %}{{ user `http_proxy` }}{% endraw %}",
+            "HTTPS_PROXY={% raw %}{{ user `https_proxy` }}{% endraw %}",
+            "http_proxy={% raw %}{{ user `http_proxy` }}{% endraw %}",
+            "https_proxy={% raw %}{{ user `http_proxy` }}{% endraw %}",
```
Comment (mattburgess) [inline dw-general-ami/generic_packer_template.json.j2]: likewise, `https_proxy` in the inner backticks

**Pull request merged**